app.go

```diff
@@ -193,9 +198,43 @@ func New(opts ...Option) *App {
 		app.provide(p)
 	}
 	app.provide(func() Lifecycle { return app.lifecycle })
+
+	if app.err != nil {
+		app.logger.Printf("Error after options were applied: %v", app.err)
+		return app
+	}
+
+	if err := app.executeInvokes(); err != nil {
+		app.err = err
```

Comment:
I'd really like to be able to get at this err without having to call Start()/trigger lifecycle.
Ideas:
- fx.TryNew() method which returns (*App, error)
- fx.Option that is an error callback
- app.Error() method